src/debug/StaticWebAppDebugProvider.ts

```diff
@@ -0,0 +1,52 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken, DebugConfiguration, DebugConfigurationProvider, tasks, WorkspaceFolder } from "vscode";
+import { callWithTelemetryAndErrorHandling, IActionContext } from "vscode-azureextensionui";
+import { validateSwaCliInstalled } from '../commands/cli/validateSwaCliInstalled';
+import { tryGetApiLocations } from '../commands/createStaticWebApp/tryGetApiLocations';
+import { swaTaskLabel } from '../constants';
+import { getFunctionsApi } from '../getExtensionApi';
+import { localize } from '../utils/localize';
+import { AzureFunctionsExtensionApi } from '../vscode-azurefunctions.api';
+
+export class StaticWebAppDebugProvider implements DebugConfigurationProvider {
+    public async provideDebugConfigurations(_folder?: WorkspaceFolder, _token?: CancellationToken): Promise<DebugConfiguration[]> {
+        // Don't provide any debug configurations automatically for now
+        return []
+    }
+
+    public async resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DebugConfiguration, _token?: CancellationToken): Promise<DebugConfiguration | undefined> {
+        return await callWithTelemetryAndErrorHandling('resolveSwaDebugConfiguration', async (context: IActionContext) => {
+            if (debugConfiguration.preLaunchTask === swaTaskLabel) {
+                if (!folder) {
+                    return;
+                }
+                const hasSwaCli = await validateSwaCliInstalled(context, localize('installSwaCli', 'You must have the Azure Static Web Apps CLI installed to debug your static web app.'));
+                if (!hasSwaCli) {
+                    return undefined;
+                }
+
+                const apiLocations = await tryGetApiLocations(context, folder);
+                const hasApi: boolean = !!apiLocations && apiLocations.length > 0;
+
+                if (hasApi) {
```

Comment:
It's just a style thing so what you've written is perfectly valid, but another way to do it is this (which would eliminate the `hasApi` variable)
```suggestion
                if (apiLocations?.length) {
```

Basically, if `apiLocations` is `null` or `undefined`, then that will be the expression result--both are falsy, so the `if` won't execute. If `apiLocations` is defined but length is 0, then 0 will be the expression result--also falsy.